lib/ddtrace/contrib/registerable.rb

```diff
@@ -0,0 +1,33 @@
+require 'ddtrace/registry'
+
+module Datadog
+  module Contrib
+    # Defines registerable behavior for integrations
+    module Registerable
+      def self.included(base)
+        base.send(:extend, ClassMethods)
+        base.send(:include, InstanceMethods)
+      end
+
+      # Class methods for registerable behavior
+      module ClassMethods
+        def register_as(name, options = {})
+          registry = options.fetch(:registry, Datadog.registry)
+          auto_patch = options.fetch(:auto_patch, false)
```

Comment:
Autopatch seems to not be used anymore. Is this only for backwards compatibility ? If so I think we can just not support it in the new configuration approach anymore.

Reply:
Backwards compatibility, yeah. I'm not sure whether we can remove this, since there are a lot of users who upgrade from really old versions (< `0.10.0`) and leave `auto_patch` in their configuration. Let me look deeper into this, but this is a great point for consideration.

Reply:
@pawelchcki @delner no breaking changes must be introduced that way. If there is a risk to break something at bootstrap time, we need to add a deprecation warning and decide when we're going to remove that field. Probably we can remove most of the things once we're heading to 1.0 release.

Reply:
@palazzem agreed.

However this shouldn't affect integrations that do not explicitly include new `Registerable` module. Also the user still would be able to call the same method signature `register_as :name, auto_patch: true`. Now `auto_patch` will be ignored a bit earlier.

It doesn't look like breaking change to me.

Reply:
@palazzem @palazzem Well, it seems like since we have to deprecate and remove `auto_patch` from the Registry anyways, why not leave this in for now, and deprecate/remove at the same time when we do it for the registry? Would be just as easy, and consistent.